### PR TITLE
[modelio] Update for iOS 10.2 beta 1

### PR DESCRIPTION
**Pull request by VincentDondain**

### CODE REVIEW

**Comment by dalexsoto** [comment]
LGTM 👍 just fix the tests

File: src/ModelIO/MDLNoiseTexture.cs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+namespace XamCore.ModelIO {
+
+	public enum  MDLNoiseTextureType {
+		Vector,
+		Cellular,
+	}
+
+	public partial class MDLNoiseTexture {
+		[iOS (9,0), Mac (10,11, onlyOn64 : true)]
+		public MDLNoiseTexture (float input, string name, Vector2i textureDimensions, MDLTextureChannelEncoding channelEncoding) : this (input, name, textureDimensions, channelEncoding, MDLNoiseTextureType.Vector)
```
Comment (spouliot) [inline src/ModelIO/MDLNoiseTexture.cs]: I see now that it replace an existing API :)

File: src/ModelIO/MDLNoiseTexture.cs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+		{
+		}
+
+		public MDLNoiseTexture (float input, string name, Vector2i textureDimensions, MDLTextureChannelEncoding channelEncoding, MDLNoiseTextureType type)
```
Comment (spouliot) [inline src/ModelIO/MDLNoiseTexture.cs]: more like `[iOS (10,2), Mac (10,12, onlyOn64 : true)]` then, even if the API would partially work on earlier version

File: src/ModelIO/MDLNoiseTexture.cs [hunk 3]
```diff
@@ -0,0 +1,35 @@
+#if XAMCORE_2_0 || !MONOMAC
+using System;
+using XamCore.ObjCRuntime;
+using Vector2i = global::OpenTK.Vector2i;
+
+namespace XamCore.ModelIO {
+
+	public enum  MDLNoiseTextureType {
+		Vector,
+		Cellular,
```
Comment (spouliot) [inline src/ModelIO/MDLNoiseTexture.cs]: add `[iOS (9,0), Mac (10,11, onlyOn64 : true)]`

File: src/modelio.cs [hunk 4]
```diff
@@ -1929,6 +1963,9 @@ interface MDLVertexBufferLayout : NSCopying
 	[iOS (9,0)][Mac(10,11, onlyOn64 : true)]
 	[BaseType (typeof (NSObject))]
 	interface MDLSubmeshTopology {
+		[Export ("initWithSubmesh:")]
+		IntPtr Constructor (MDLSubmesh submesh);
```
Comment (spouliot) [inline src/modelio.cs]: missing availability attributes

File: src/modelio.cs [hunk 5]
```diff
@@ -1739,8 +1775,6 @@ interface MDLVertexDescriptor : NSCopying
 	interface MDLVoxelArray
 	{
 
-		[Deprecated (PlatformName.MacOSX, 10, 12, message: "Use new MDLVoxelArray (MDLAsset, int, float)")]
-		[Obsoleted (PlatformName.iOS, 10, 0, message: "Use new MDLVoxelArray (MDLAsset, int, float)")]
```
Comment (spouliot) [inline src/modelio.cs]: ^ I don't see this removal in the diff ?

**Pull request merged**